### PR TITLE
fix(broker): send server name with SSL handshake

### PR DESCRIPTION
**Pull request by brentos**
Fixes an issue where handshake fails when there are proxies in front of the Kafka broker. This happens when using VPC private links in AWS, for example.

### CODE REVIEW

File: dev/openresty/Dockerfile
Comment (jschmid1) [inline dev/openresty/Dockerfile:28]: This shouldn't pin 3.5. Either we move with the latest `stable` tag if master is too aggressive or have separate release branches that are tied to Kong versions.

**Review by jschmid1 — APPROVED** [review]
Other than the version pinning, lgtm

**Review by jschmid1 — CHANGES_REQUESTED** [review]
meant to _request changes_

File: dev/openresty/Dockerfile
Comment (brentos) [reply dev/openresty/Dockerfile]: Forgot to remove this commented line.

File: dev/docker-compose.yaml
Comment (outsinre) [inline dev/docker-compose.yaml:214]:
```suggestion
    image: kong/kong-gateway:latest
```

1. This repo only serves Kong EE.
2. Should we pin to `3.6`? By default, this repo represents its latest and maybe test against the latest Kong EE is better.

File: spec/sasl_ssl_spec.lua
Comment (outsinre) [inline spec/sasl_ssl_spec.lua:65]:
```suggestion
```

This is not wanted.

File: dev/docker-compose.yaml
Comment (brentos) [reply dev/docker-compose.yaml:214]: I just needed a TCP proxy that supports routing by SNI and TLS passthrough, so really it shouldn't matter for my test case. My only concern with using `kong-gateway:latest` is if something in Kong EE breaks that functionality then this test will begin to fail even if this library is still functioning correctly.

File: spec/sasl_ssl_spec.lua
Comment (brentos) [reply spec/sasl_ssl_spec.lua:65]: I want to test here with the TCP proxy, which is why I create a new client that points to the proxy instead of the actual broker.

**Review by outsinre — APPROVED** [review]

**Comment by outsinre** [comment]
I think this can be merged.

**Review by gszr — APPROVED** [review]
good to go; once merged, please create a new GH release and pin on both kafka plugins

**Comment by outsinre** [comment]
Will bump version.

**Pull request merged**
